pandas-stubs/core/interchange/dataframe_protocol.pyi

```diff
@@ -0,0 +1,117 @@
+import abc
+from abc import (
+    ABC,
+    abstractmethod,
+)
+import enum
+from typing import (
+    Any,
+    Iterable,
+    Sequence,
+    TypedDict,
+)
+
+class DlpackDeviceType(enum.IntEnum):
+    CPU: int
+    CUDA: int
+    CPU_PINNED: int
+    OPENCL: int
+    VULKAN: int
+    METAL: int
+    VPI: int
+    ROCM: int
+
+class DtypeKind(enum.IntEnum):
+    INT: int
+    UINT: int
+    FLOAT: int
+    BOOL: int
+    STRING: int
+    DATETIME: int
+    CATEGORICAL: int
+
+class ColumnNullType(enum.IntEnum):
+    NON_NULLABLE: int
+    USE_NAN: int
+    USE_SENTINEL: int
+    USE_BITMASK: int
+    USE_BYTEMASK: int
+
+class ColumnBuffers(TypedDict):
+    data: tuple[Buffer, Any]
+    validity: tuple[Buffer, Any] | None
+    offsets: tuple[Buffer, Any] | None
+
+class CategoricalDescription(TypedDict):
+    is_ordered: bool
+    is_dictionary: bool
+    categories: Column | None
+
+class Buffer(ABC, metaclass=abc.ABCMeta):
+    @property
+    @abstractmethod
+    def bufsize(self) -> int: ...
+    @property
+    @abstractmethod
+    def ptr(self) -> int: ...
+    @abstractmethod
+    def __dlpack__(self): ...
+    @abstractmethod
+    def __dlpack_device__(self) -> tuple[DlpackDeviceType, int | None]: ...
+
+class Column(ABC, metaclass=abc.ABCMeta):
+    @property
+    @abstractmethod
+    def size(self) -> int: ...
```

Comment:
Looking at https://data-apis.org/dataframe-protocol/latest/API.html, returns `Optional[int]`

Reply:
I think the spec type might also be wrong here.  The text indicates that the size is required, either the full DF size or the chunk size if a single chunk.